core/i18n/src/ITwinLocalization.ts

```diff
@@ -143,25 +143,20 @@ export class ITwinLocalization implements Localization {
     return value;
   }
 
-  /** Similar to `getLocalizedString` but the namespace is a separate param and the key does not include the namespace.
+  /** Similar to `getLocalizedString` but the default namespace is a separate parameter and the key does not need
```

Comment:
Instead of changing the functionality of this method. Let's just remove it. It's marked as internal and we should clean this up since no one should be using it.

FYI @raplemie  @GerardasB this is used in AppUI so we'll be updating it to use the correct method. The same changes will need to be made for 4.0.

Reply:
Added https://github.com/iTwin/itwinjs-backlog/issues/510 for tracking

Reply:
I've pushed a commit that reverts changes to `getLocalizedStringWithNamespace` and skips a few edge-case tests that fail without the better code. This will preserve the method to make sure that it will work (because no changes were made) up until it is removed. The removal of the method will be in a separate PR as it is a bigger and breaking change - leaving this PR focused on tests - and is tracked by [this issue](https://github.com/iTwin/itwinjs-backlog/issues/513).